### PR TITLE
Promote QR Image in Terraform Repo SaaS File

### PR DESCRIPTION
**Pull request by cubismod**
[SDE-3686](https://issues.redhat.com/browse/SDE-3686)

Dependent on this MR: https://gitlab.cee.redhat.com/service/app-interface/-/merge_requests/120268

Includes Terraform Repo in the auto promotion process for Qontract Reconcile

### CODE REVIEW

File: reconcile/utils/mr/promote_qontract.py [hunk 1]
```diff
@@ -184,6 +184,15 @@ def process(self, gitlab_cli: GitLabApi) -> None:
             replace_text=self.commit_sha,
         )
 
+        # data/services/app-interface/terraform-repo/cicd/ci-int/saas-terraform-repo.yaml
+        self._process_by(
+            "json_path",
+            gitlab_cli=gitlab_cli,
+            path="data/services/app-interface/terraform-repo/cicd/ci-int/saas-terraform-repo.yaml",
+            search_text="$.resourceTemplates[?(@.url == 'https://gitlab.cee.redhat.com/app-sre/terraform-repo-tekton')].targets[?(@.name == 'tf-repo-prod')].parameters.QR_IMAGE_TAG",
```
Comment (hemslo) [inline reconcile/utils/mr/promote_qontract.py:192]: there are 2 matches for url `https://gitlab.cee.redhat.com/app-sre/terraform-repo-tekton`, would it be cleaner to use `name` field `terraform-repo`?

**Review by hemslo — APPROVED** [review]
LGTM!

File: reconcile/utils/mr/promote_qontract.py (hunk 1 above)
Comment (cubismod) [reply reconcile/utils/mr/promote_qontract.py]: Nice find! Fixed.

**Review by maorfr — COMMENTED** [review]
:heart:

**Pull request merged**